src/Microsoft.AspNetCore.Mvc.DataAnnotations/Internal/DataAnnotationsMetadataProvider.cs

```diff
@@ -80,23 +89,55 @@ public void CreateDisplayMetadata(DisplayMetadataProviderContext context)
                 displayMetadata.DataTypeName = DataType.Html.ToString();
             }
 
-            // Description
+            // DisplayFormatString
+            if (displayFormatAttribute != null)
+            {
+                displayMetadata.DisplayFormatString = displayFormatAttribute.DataFormatString;
+            }
+
+            // DisplayDescription
             if (displayAttribute != null)
             {
-                displayMetadata.Description = () => displayAttribute.GetDescription();
-                displayMetadata.Placeholder = () => displayAttribute.GetPrompt();
+                if (localizer != null &&
+                    !string.IsNullOrEmpty(displayAttribute.Name) &&
+                    displayAttribute.ResourceType == null)
```

Comment:
Need a bit more testing here. Tests cover the `localizer` part of this condition but not the `ResourceType` part. (`Description` part doesn't matter since default resource name (`"Description"`) isn't used unless `ResourceType != null`.) Same for test coverage of `Prompt` and `Name`.